solidity/contracts/middleware/InterchainAccountRouter.sol

```diff
@@ -69,119 +145,317 @@ contract InterchainAccountRouter is Router, IInterchainAccountRouter {
-     * @param calls The sequence of calls to be relayed.
+     * @notice Registers the address of many remote InterchainAccountRouter
+     * contracts to use as a default when making interchain calls
+     * @param _destinations The remote domains
+     * @param _routers The addresses of the remote InterchainAccountRouters
+     */
+    function enrollRemoteRouters(
+        uint32[] calldata _destinations,
+        bytes32[] calldata _routers
+    ) external {
+        require(_destinations.length == _routers.length, "!length");
+        for (uint256 i = 0; i < _destinations.length; i += 1) {
+            enrollRemoteRouter(_destinations[i], _routers[i]);
+        }
+    }
+
+    /**
+     * @notice Registers the address of remote InterchainAccountRouter
+     * and ISM contracts to use as a default when msg.sender makes
+     * interchain calls
+     * @param _destination The remote domain
+     * @param _router The address of the remote InterchainAccountRouter
+     * @param _ism The address of the remote ISM
+     */
+    function overrideRemoteRouterAndIsm(
+        uint32 _destination,
+        bytes32 _router,
+        bytes32 _ism
+    ) external {
+        OverridableEnumerableMap.setOverride(
+            routersMap,
+            msg.sender,
+            _destination,
+            _router
+        );
+        OverridableEnumerableMap.setOverride(
+            ismsMap,
+            msg.sender,
+            _destination,
+            _ism
+        );
+        emit RemoteRouterOverridden(_destination, _router, msg.sender);
+        emit RemoteIsmOverridden(_destination, _ism, msg.sender);
+    }
+
+    /**
+     * @notice Dispatches a sequence of remote calls to be made by a owner's
+     * interchain account on the destination domain
+     * @dev Uses the default router and ISM addresses for the destination
+     * domain, reverting if none have been configured
      * @dev Recommend using CallLib.build to format the interchain calls.
+     * @param _destination The remote domain of the chain to make calls on
+     * @param _calls The sequence of calls to make
+     * @return The Hyperlane message ID
      */
-    function dispatch(uint32 _destinationDomain, CallLib.Call[] calldata calls)
+    function callRemote(uint32 _destination, CallLib.Call[] calldata _calls)
         external
         returns (bytes32)
     {
+        (bytes32 _router, bytes32 _ism) = getRemoteRouterAndIsm(
+            _destination,
+            msg.sender
+        );
+        require(_router != bytes32(0), "no router specified for destination");
+        return callRemoteWithOverrides(_destination, _router, _ism, _calls);
+    }
+
+    /**
+     * @notice Handles dispatched messages by relaying calls to the interchain account
+     * @param _origin The origin domain of the interchain account
+     * @param _sender The sender of the interchain message
+     * @param _message The InterchainAccountMessage containing the account
```

Comment:
maybe @dev that it doesn't need to be onlyRemoteRouter

Reply:
Good call